builder-base/Dockerfile

```diff
@@ -194,7 +195,29 @@ RUN --mount=type=cache,target=/var/cache/yum,sharing=locked \
     /install_bash.sh && \
     /remove_yum_packages.sh
 
-FROM ${BUILDER_IMAGE} as ansible
+# for al2 builds we pull python3.9 from the minimal image build since al2
+# does not ship 3.9
+FROM ${PYTHON_IMAGE} as python-3.9
+ARG ANSIBLE_VERSION
+ENV ANSIBLE_VERSION=$ANSIBLE_VERSION
+ARG PYWINRM_VERSION
+ENV PYWINRM_VERSION=$PYWINRM_VERSION
+RUN python3 -m ensurepip --upgrade && \
+    pip3 install --no-cache-dir ansible-core==$ANSIBLE_VERSION && \
+    pip3 install --no-cache-dir pywinrm==$PYWINRM_VERSION
+
+FROM ${BUILDER_IMAGE} as ansible-al2
+
+COPY --link --from=python-3.9 / /ansible
+# we want python2 to be the version symlinkd to python
+# we also do not want to overwrite the rpm db
+# this will leave the packages pulled in from the python image in
+# a slightly odd state in that they are installed yet not in the rpm db
+RUN set -x && \
+    unlink /ansible/usr/bin/python && \
+    rm -rf /var/lib/
+
+FROM ${BUILDER_IMAGE} as ansible-al23
```

Comment:
for the al23 base image continue doing what we were since that version includes python 3.9